### DISCUSSION TITLE
Help Building niri on void linux

### DISCUSSION THREAD

**Discussion by tomysurya** [YaLTeR/niri#423] [Q&A]
hello guys, i try to build niri on void linux, at the final step of building niri i get this error:
note: /usr/bin/ld: cannot find -linput: nosuch file or directory
         collect2: error: ld returned 1 exit status
what package did i missed?
appreciate any help thank you!

**Comment by kchibisov** [comment]
You need `libinput`.

**Comment by tomysurya** [comment]
okay i'll try thank you

**Comment by tomysurya** [comment]
the build was succesfull but whenever i run it from tty my monitor was turned off then i press any keys or move mouse theres nothing happens, the monitor keeps off

**Reply by YaLTeR** [reply]
Can you post the full niri output?

**Reply by tomysurya** [reply]
how can i get them? sorry total noobs here

**Reply by YaLTeR** [reply]
Something like `niri >niri.log 2>&1`

**Reply by tomysurya** [reply]
when i type niri on my tty the monitor goes off i can't do nothing

**Reply by YaLTeR** [reply]
When that happens, press Super+Shift+E then Enter to exit niri and send the log file here

**Reply by rrveex** [reply]
Here's a working (for me) template for niri

```
# Template file for 'niri'
pkgname=niri
version=0.1.7
revision=1
build_style=cargo
build_helper=qemu
hostmakedepends="pkgconf clang"
makedepends="eudev-libudev-devel libxkbcommon-devel libinput-devel libseat-devel cairo-devel pango-devel pipewire-devel libgbm-devel"
# depends="libpipewire pango cairo eudev-libudev libinput pixman" ???
short_desc="A scrollable-tiling Wayland compositor."
maintainer="Orphaned <orphan@voidlinux.org>"
license="GPL-3.0"
homepage="https://github.com/YaLTeR/niri"
changelog="https://github.com/YaLTeR/niri/releases/tag/v0.1.7"
distfiles="https://github.com/YaLTeR/niri/archive/refs/tags/v${version}.tar.gz"
checksum=2ce7a450b550164f99d6fd6d2815c04f2160fbf932f9c0d244b7b1e3481c9b21

post_install() {
	vinstall resources/niri.desktop 644 usr/share/wayland-sessions
	vinstall resources/niri-portals.conf 644 usr/share/xdg-desktop-portal
}
```
I don't know what is needed in `depends`, because I already have stuff installed, so it "just works" on my machine.

Since you mentioned missing libraries: you can do something like:
```
ldd -r `which niri`
```
If it says something "not found", do a `xlocate` on that library to find the package providing it, add this to `depends`. Build again, `xi -f niri`, try again.
